airtable/plugin.py

```diff
@@ -30,16 +38,15 @@ class AirtableRecordID:
 
 def airtable_parse_ids_from_url(
     url: str,
-) -> Tuple[AirtableBaseID, Optional[AirtableTableID], Optional[AirtableRecordID]]:
+) -> Tuple[AirtableBaseID, Optional[AirtableTableID], Optional[AirtableRecordID], Optional[AirtableViewID]]:
```

Comment:
Did you try the idea of having a super-object that contained all the sub IDs instead of returning the tuple?

I feel like this exposes a lot of the detail to code gen to properly handle and pass all of the individual elements, and maybe it's better if the plumbing is simple and all APIs take AirtableID and our implementations resolve how to use it?

Reply:
Yeah, the first version I tried was
```
@dataclass
class AirtableDataContextID:
    base_id: str
    table_id: Optional[str]
    view_id: Optional[str]

def airtable_parse_ids_from_url(
    url: str
) -> Tuple[AirtableDataContextID, Optional[AirtableRecordID]:
    ...

def airtable_record_list(
    context_id: AirtableDataContextID,
    include_fields: Optional[set[str]] = None,
) list[AirtableRecord]:
    ...
```

@jcharum do you see any issue with the version with returning an object that contains the base_id, table_id, view_id? I think a variation of this was discuss before but we landed on the version that returns the ids separately.

Reply:
I have not done a systematic comparison, but that seems worse to me as it moves something that is type-checked into something that isn't.

Reply:
To close the loop out:

```
so the issue is that some APIs require specific sets of IDs





[3:25](https://d8e-ai.slack.com/archives/D05ABC70YR0/p1707866748928529)
e.g., some APIs only take a baseid
[3:25](https://d8e-ai.slack.com/archives/D05ABC70YR0/p1707866756661279)
others take one with a record ID
[3:26](https://d8e-ai.slack.com/archives/D05ABC70YR0/p1707866769272949)
and if you have a superclass, the type doesn't define what it contains and whether it's relevant
[3:26](https://d8e-ai.slack.com/archives/D05ABC70YR0/p1707866776661009)
which is something code gen probably does need to know about
[3:27](https://d8e-ai.slack.com/archives/D05ABC70YR0/p1707866822332539)
one alternative would be some kind of "discriminated union" type, but code gen would need to check the type of the ID, which MAY be better than optionals
[3:27](https://d8e-ai.slack.com/archives/D05ABC70YR0/p1707866864801369)
but it's not clear, so probably not worth changing significantly right now
```